agent/src/beerocks/slave/son_slave_thread.cpp

```diff
@@ -3862,77 +3862,44 @@ bool slave_thread::handle_autoconfiguration_renew(Socket *sd, ieee1905_1::CmduMe
 bool slave_thread::handle_autoconfiguration_wsc(Socket *sd, ieee1905_1::CmduMessageRx &cmdu_rx)
 {
     LOG(DEBUG) << "Received AP_AUTOCONFIGURATION_WSC_MESSAGE";
+    // According to section 17.1.3 in the Multi-AP Specification version 1.0,
+    // if an AP-Autoconfiguration WSC message is sent by the Multi-AP Agent,
+    // it includes one AP Radio basic capabilities TLV and one WSC TLV (containing M1).
+    // If the the message is sent by the Multi-AP controller, it includes one
+    // AP radio Identifier TLV and one or more WSC TLVs (containing M2).
+    // Check if this is the case and this is an M1 message we sent to the controller
+    // which was just looped back (when we send the M1 to the local controller,
+    // it will be published back on the local bus because the destination is our
+    // AL-MAC, and the agent does listen to this CMDU message type).
+    // If this is a looped back M1 CMDU, we can treat is as handled successfully.
```

Comment:
Actually, we could just drop this special handling entirely now. The return value is anyway ignored (it gets passed up to socket_thread::work without further comment, and there it essentially gets ignored). We really should be checking for M1/M2, which we do implicitly by checking m2_list.empty().